src/autocapture-utils.ts

```diff
@@ -97,13 +98,47 @@ export const usefulElements = ['a', 'button', 'form', 'input', 'select', 'textar
  * using a variety of heuristics.
  * @param {Element} el - element to check
  * @param {Event} event - event to check
+ * @param {Object} autocaptureConfig - autocapture config
  * @returns {boolean} whether the event should be captured
  */
-export function shouldCaptureDomEvent(el: Element, event: Event): boolean {
+export function shouldCaptureDomEvent(
+    el: Element,
+    event: Event,
+    autocaptureConfig: AutocaptureConfig | undefined = undefined
+): boolean {
     if (!el || isTag(el, 'html') || !isElementNode(el)) {
         return false
     }
 
+    if (autocaptureConfig?.url_allowlist) {
+        const url = window.location.href
+        const allowlist = autocaptureConfig.url_allowlist
+        if (allowlist && !allowlist.some((regex) => url.match(regex))) {
```

Comment:
Since we're very performance conscious with the SDK maybe we should be precompiling these regexes.

Might not make much difference since you'd expect we run this code thousands of times not millions but....

Reply:
Good point, will do

Reply:
done :)

Reply:
Realized that it's already compiled if they are passing in a regex but also enabled them to pass in a string, as in the example above